sdks/beam-transforms/src/main/java/org/gbif/pipelines/transforms/core/MultiTaxonomyTransform.java

```diff
@@ -0,0 +1,112 @@
+ *
+ * <p>ParDo runs sequence of interpretations for {@link MultiTaxonRecord} using {@link
+ * ExtendedRecord} as a source and {@link MultiTaxonomyInterpreter} as interpretation steps
+ *
+ * @see <a href="https://dwc.tdwg.org/terms/#taxon</a>
+ */
+@Slf4j
+public class MultiTaxonomyTransform extends Transform<ExtendedRecord, MultiTaxonRecord> {
+
+  private final SerializableSupplier<KeyValueStore<NameUsageMatchRequest, NameUsageMatchResponse>>
+      kvStoreSupplier;
+  private KeyValueStore<NameUsageMatchRequest, NameUsageMatchResponse> kvStore;
+  private final List<String> checklistKeys;
+
+  @Builder(buildMethodName = "create")
+  private MultiTaxonomyTransform(
+      SerializableSupplier<KeyValueStore<NameUsageMatchRequest, NameUsageMatchResponse>>
+          kvStoresSupplier,
+      List<String> checklistKeys) {
+    super(
+        MultiTaxonRecord.class,
+        MULTI_TAXONOMY,
+        MultiTaxonomyTransform.class.getName(),
+        TAXON_RECORDS_COUNT);
+    this.kvStoreSupplier = kvStoresSupplier;
+    this.checklistKeys = checklistKeys;
+  }
+
+  /** Maps {@link MultiTaxonRecord} to key value, where key is {@link MultiTaxonRecord#getId} */
+  public MapElements<MultiTaxonRecord, KV<String, MultiTaxonRecord>> asKv(boolean useCoreId) {
+    return MapElements.into(new TypeDescriptor<KV<String, MultiTaxonRecord>>() {})
+        .via((MultiTaxonRecord tr) -> KV.of(useCoreId ? tr.getCoreId() : tr.getId(), tr));
+  }
+
+  /** Maps {@link MultiTaxonRecord} to key value, where key is {@link MultiTaxonRecord#getId} */
+  public MapElements<MultiTaxonRecord, KV<String, MultiTaxonRecord>> toKv() {
+    return asKv(false);
+  }
+
+  public MultiTaxonomyTransform counterFn(SerializableConsumer<String> counterFn) {
+    setCounterFn(counterFn);
+    return this;
+  }
+
+  /** Beam @Setup initializes resources */
+  @Setup
+  public void setup() {
+    if (kvStore == null && kvStoreSupplier != null) {
+      log.info("Initialize NameUsageMatchKvStores");
+      kvStore = kvStoreSupplier.get();
+    }
+  }
+
+  /** Beam @Setup can be applied only to void method */
+  public MultiTaxonomyTransform init() {
+    setup();
+    return this;
+  }
+
+  /** Beam @Teardown closes initialized resources */
+  @Teardown
+  public void tearDown() {
+    if (kvStore != null) {
+      log.info("Close NameUsageMatchKvStores");
+      try {
+        kvStore.close();
+      } catch (IOException ex) {
+        log.error("Error closing KV Store", ex);
+      }
+    }
+  }
+
+  @Override
+  public Optional<MultiTaxonRecord> convert(ExtendedRecord source) {
+    return Interpretation.from(source)
+        .to(MultiTaxonRecord.newBuilder().setCreated(Instant.now().toEpochMilli()).build())
+        .when(er -> !er.getCoreTerms().isEmpty())
+        .via(MultiTaxonomyInterpreter.taxonomyInterpreter(kvStore, checklistKeys))
+        .via(MultiTaxonomyInterpreter::setCoreId)
+        .via(MultiTaxonomyInterpreter::setParentEventId)
```

Comment:
can't the setCoreId and setParentEventId done in the MultiTaxonomyInterpreter.taxonomyInterpreter? also the MultiTaxonomyInterpreter.taxonomyInterpreter shouldn't be called MultiTaxonomyInterpreter.interpretMultiTaxonomy or sth like that?

Reply:
done